### PR TITLE
odo registry -o json

### LINKED ISSUE #5644: Provide an `odo registry -o json` command

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind feature

**What does this PR do / why we need it:**

- [x] ref doc for `odo registry`
- [x] odo registry -o json
- [x] ref doc for `odo registry -o json`
- [x] Move structures used for JSON output to `api` package

**Which issue(s) this PR fixes:**

Fixes #5644

**PR acceptance criteria:**

- [x] Unit test 

- [x] Integration test 

- [x] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: docs/website/versioned_docs/version-3.0.0/command-reference/registry.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+## Available Flags
+
+By default, `odo registry` lists all the Devfile stacks from all the Devfile registries.
+
+These flags let you filter the listed Devfile stacks:
+
+* `--devfile <name>` to list the Devfile stacks with this exact name
+* `--devfile-registry <name>` to list the Devfile stack of this registry (this is the `name` used
+when adding the registry to the preferences with `odo preference registry add <name> <url>`)
+* `--filter <term>` to list the Devfile for which the term is found in the devfile name or description
```
Comment (rm3l) [inline docs/website/versioned_docs/version-3.0.0/command-reference/registry.md:20]: Can you also update the CLI Help message of this command? Currently, it says that the filter is based on the name of the component:

```
❯ odo help registry        
List all components from the Devfile registry

Usage:
  odo registry [flags]

Flags:
      --details                   Show details of each component
      --devfile string            Only the specific Devfile component
      --devfile-registry string   Only show components from the specific Devfile registry
      --filter string             Filter based on the name of the component
```

File: docs/website/versioned_docs/version-3.0.0/command-reference/json-output.md [hunk 2]
```diff
@@ -156,16 +156,14 @@ The `odo list` command returns information about components running on a specifi
 
 The `components` field lists the components either deployed in the cluster, or defined in the local Devfile.
 
-The `componentsInDevfile` field lists the names of components present in the `components` list that are defined in the local Devfile.
+The `componentInDevfile` field gives the name of components present in the `components` list that are defined in the local Devfile, or is empty if no local Devfile is present.
```
Comment (rm3l) [inline docs/website/versioned_docs/version-3.0.0/command-reference/json-output.md:159]: I guess `components` should not be plural here, as we are now returning at most one name.

```suggestion
The `componentInDevfile` field gives the name of the component present in the `components` list and defined in the local Devfile, or is empty if no local Devfile is present.
```

**Review by rm3l — CHANGES_REQUESTED** [review]
Overall LGTM. I added few comments however ;)

**Review by rm3l — COMMENTED** [review]
Another small improvement about the error message if a wrong registry is specified.

Wouldn't it be better, from the user perspective, to report that the registry was not found in the registry list, rather than saying `no deployable components found`?

```
❯ odo registry --devfile java-maven --devfile-registry fake-registry -o json
{
        "message": "no deployable components found"
}
```

File: pkg/api/registry.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	DisplayName     string       `json:"displayName"`
+	Description     string       `json:"description"`
+	Registry        Registry     `json:"registry"`
+	Language        string       `json:"language"`
+	Tags            []string     `json:"tags"`
+	ProjectType     string       `json:"projectType"`
+	Version         string       `json:"version"`
+	StarterProjects []string     `json:"starterProjects"`
+	DevfileData     *DevfileData `json:"devfileData,omitempty"`
+}
```
[the remark this answers is not in the document]
Comment (feloy) [reply pkg/api/registry.go:24]: I don't think anymore this information is necessary. Removing it from the Issue

**Review by rm3l — APPROVED** [review]

File: tests/integration/devfile/cmd_devfile_init_test.go [hunk 4]
```diff
@@ -125,7 +125,7 @@ var _ = Describe("odo devfile init command tests", func() {
 				Expect(stderr).To(BeEmpty())
 				Expect(helper.IsJSON(stdout)).To(BeTrue())
 				helper.JsonPathContentIs(stdout, "devfilePath", filepath.Join(commonVar.Context, "devfile.yaml"))
-				helper.JsonPathContentIs(stdout, "devfileData.devfile.schemaVersion", "2.0.0")
+				helper.JsonPathContentIs(stdout, "devfileData.devfile.metadata.name", compName)
```
Comment (rm3l) [inline tests/integration/devfile/cmd_devfile_init_test.go:128]: I was about to check why tests were not passing, but you were too faster :-P 👍🏿

**Review by rm3l — APPROVED** [review]

**Comment by feloy** [comment]
```
• Failure [7.628 seconds]
create/delete/list/get/set namespace tests
/go/src/github.com/redhat-developer/odo/tests/integration/cmd_namespace_test.go:14
  when using the alias namespace to delete a namespace
  /go/src/github.com/redhat-developer/odo/tests/integration/cmd_namespace_test.go:52
    should successfully delete the namespace using the force flag and waiting [It]
    /go/src/github.com/redhat-developer/odo/tests/integration/cmd_namespace_test.go:81
```

/test v4.10-integration-e2e

**Comment by feloy** [comment]
Flaky tests on non related tests
Tests pass on IBM Cloud

/override ci/prow/v4.10-integration-e2e

**Pull request merged**